src/components/NewsCardArticles/NewsCardArticles.jsx

```diff
@@ -0,0 +1,31 @@
+import './NewsCardArticles.css';
+import NewsCard from '../NewsCard/NewsCard';
+import { useLocation } from 'react-router-dom';
+
+ function NewsCardArticles({
+    cardList, newsArticlesCounts, savedNewsArticles = [], handleSavedNewsArticles, handleDeleteNewsArticles,
+}) {
+    const location = useLocation();
+    const isNewsArticlesSavedPage = location.pathname === "/saved-news";
+
+    const displayedNewsArticles = isNewsArticlesSavedPage ? savedNewsArticles : cardList.slice(0, newsArticlesCounts);
+    console.log(cardList);
+    console.log(newsArticlesCounts);
+
+    return (
+        <div className="articles__selected">
+            {displayedNewsArticles.map((item, index) => {
+                return (
+            <NewsCard 
+            handleSavedNewsArticles={handleSavedNewsArticles}
+            handleDeleteNewsARticles={handleDeleteNewsArticles}
+            item={item}
+            key={index}
+            />
+            );
+            })}
+        </div>
```

Comment:
You need to **format** the markup (and code) to show **indentation** and to make the code more readable. You can do that with `Prettier` or a combination: press `SHIFT + ALT + F`. **Please, correct it everywhere in the code**


```jsx


  return (
    <div className="articles__selected">
      {displayedNewsArticles.map((item, index) => {
        return (
          <NewsCard
            handleSavedNewsArticles={handleSavedNewsArticles}
            handleDeleteNewsARticles={handleDeleteNewsArticles}
            item={item}
            key={index}
          />
        );
      })}
    </div>
  );
```